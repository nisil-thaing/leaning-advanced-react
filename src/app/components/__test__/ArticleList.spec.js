import React from 'react';
import renderer from 'react-test-renderer';

import ArticleList from '../ArticleList';

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: { id: 'a' },
      b: { id: 'b' },
      c: { id: 'c' }
    },
    articleActions: {
      getAuthorByAuthorId: jest.fn(() => ({}))
    }
  };

  it('renderer correctly', () => {
    const tree  = renderer.create(
      <ArticleList
        { ...testProps }
      />
    ).toJSON();

    expect(tree.children.length).toBe(3);

    expect(tree).toMatchSnapshot();
  });
});
