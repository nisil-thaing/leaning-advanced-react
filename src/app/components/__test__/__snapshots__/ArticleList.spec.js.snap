// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renderer correctly 1`] = `
<div>
  <div
    className="Article"
  >
    <div
      className="title"
    />
    <div
      className="timestamp"
    >
      Invalid Date
    </div>
    <div
      className="author"
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      className="body"
    />
  </div>
  <div
    className="Article"
  >
    <div
      className="title"
    />
    <div
      className="timestamp"
    >
      Invalid Date
    </div>
    <div
      className="author"
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      className="body"
    />
  </div>
  <div
    className="Article"
  >
    <div
      className="title"
    />
    <div
      className="timestamp"
    >
      Invalid Date
    </div>
    <div
      className="author"
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      className="body"
    />
  </div>
</div>
`;
